### ISSUE TITLE
DICOMweb with multiple load-balanced Orthanc instances

### ISSUE THREAD

**Issue by Diegovictorbr** [nroduit/Weasis#264]
Hello!

I am trying to setup Weasis to work via DICOMweb with 2 load-balanced Orthanc instances. They are served by NGINX under the URI **$HOST/orthanc-nightly-reader**

Inside Weasis, I created the configuration like this:

![image](https://user-images.githubusercontent.com/19496924/166563625-80eb1cb2-a802-4da6-8ad0-97fcbf627daa.png)

And when I query the server, I have the list of studies. However, when I try to import them, nothing happens. Weasis log shows:

03.05.2022 21:39:29.500 *DEBUG* [Dicom Q/R task-pool-2-thread-1] org.weasis.dicom.qr.RetrieveTask: QIDO-RS request: http://localhost/orthanc-nightly-reader/dicom-web/studies/1.2.392.200036.9125.2.1631116563610.64944063984.261631/series?includefield=0008103E,00080060,0020000E,00200011,00081190

Notice the URL in the first request: **http://localhost/orthanc-nightly-reader/dicom-web**

The second request, however, changes the URL and fails:

org.weasis.dicom.qr.RetrieveTask: QIDO-RS request: https://localhost/dicom-web/studies/1.2.392.200036.9125.2.1631116563610.64944063984.261631/series/1.2.392.200036.9125.3.1631116563610.64944063984.261633/instances?includefield=00080018,00200013,00081190

org.weasis.dicom.qr.RetrieveTask: QIDO-RS all instances with seriesUID 1.2.392.200036.9125.3.1631116563610.64944063984.261633 org.weasis.core.util.StreamIOException: java.net.ConnectException: Connection refused: no further information
	at org.weasis.core.api.util.NetworkUtil.prepareConnection(NetworkUtil.java:162)
	at org.weasis.core.api.util.NetworkUtil.getHttpResponse(NetworkUtil.java:80)
	at org.weasis.core.api.util.NetworkUtil.getHttpResponse(NetworkUtil.java:73)

The original URL (http://localhost/orthanc-nightly-reader/dicom-web) becomes https://localhost/dicom-web 

Is there a way to solve this so that I can serve Weasis with multiple load-balanced viewers?

OS: Windows
Java: 1.8
Weasis version: V4.0.0-RC

**Comment by nroduit** [comment]
I'm not very familiar with Orthanc but I think the difference in URL comes from the fact that it is the server that [can give the retrieved URL for an instance](https://github.com/nroduit/Weasis/blob/master/weasis-dicom/weasis-dicom-explorer/src/main/java/org/weasis/dicom/explorer/rs/RsQueryResult.java#L491-L494) (Tag.RetrieveURL).

The Nginx configuration has to respect the Orthanc configuration contexts. If Orthanc returns retrieve URLs, these must be accessible.

**Comment by Diegovictorbr** [comment]
> I'm not very familiar with Orthanc but I think the difference in URL comes from the fact that it is the server that [can give the retrieved URL for an instance](https://github.com/nroduit/Weasis/blob/master/weasis-dicom/weasis-dicom-explorer/src/main/java/org/weasis/dicom/explorer/rs/RsQueryResult.java#L491-L494) (Tag.RetrieveURL).
> 
> The Nginx configuration has to respect the Orthanc configuration contexts. If Orthanc returns retrieve URLs, these must be accessible.

Thanks for the info @nroduit , I'll get in touch with the Orthanc team and come back.

**Comment by Diegovictorbr** [comment]
Hello.

The problem indeed was related to Orthanc building the retrieve URL (https://groups.google.com/g/orthanc-users/c/Z-GYyBiBHp4). The Orthanc team quickly created a fix by adding a new config option to the DICOMweb section of the Orthanc configuration file (commonly orthanc.json).

My previous configuration for the reader instances was:

```
"DicomWeb": {
    "Enable": true, // Whether DICOMweb support is enabled
    "Root": "/dicom-web/", // Root URI of the DICOMweb API (for QIDO-RS, STOW-RS and WADO-RS)
    "EnableWado": true, // Whether WADO-URI (previously known as WADO) support is enabled
    "WadoRoot": "/wado", // Root URI of the WADO-URI (aka. WADO) API
    "StudiesMetadata": "MainDicomTags",
    "Ssl": true,
    "QidoCaseSensitive": false // For QIDO-RS server, whether search is case sensitive (since release 0.5)
}
```

And now it is:

```
"DicomWeb": {
    "Enable": true, // Whether DICOMweb support is enabled
    "Root": "/dicom-web/", // Root URI of the DICOMweb API (for QIDO-RS, STOW-RS and WADO-RS)
    "EnableWado": true, // Whether WADO-URI (previously known as WADO) support is enabled
    "WadoRoot": "/wado", // Root URI of the WADO-URI (aka. WADO) API
    "StudiesMetadata": "MainDicomTags",
    "PublicRoot": "/orthanc-nightly-reader/dicom-web/", <---------------
    "Ssl": true,
    "QidoCaseSensitive": false // For QIDO-RS server, whether search is case sensitive (since release 0.5)
}

```

I think the fix will be shipped in the version 1.8 of the Orthanc dicom-web plugin.
